calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -171,7 +171,7 @@ func unmarshalResource(tm unversioned.TypeMetadata, b []byte) ([]unversioned.Res
 		return nil, err
 	}
 
-	if err = yaml.Unmarshal(b, unpacked); err != nil {
+	if err = yaml.UnmarshalStrict(b, unpacked); err != nil {
```

Comment:
are we utilizing the special error(`UnmarshalUnknownFieldError`) this func provides to print a specific error? Just wondering where that happens since I don't see a special case that matches `UnmarshalUnknownFieldError` to print a specific message in this PR